### PR TITLE
Support for rollover index API

### PR DESCRIPTION
**Pull request by gmarz**
https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html

### CODE REVIEW

File: src/Nest/Indices/IndexManagement/RolloverIndex/RolloverIndexResponse.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Nest
+{
+	[JsonObject(MemberSerialization.OptIn)]
+	public interface IRolloverIndexResponse : IAcknowledgedResponse
+	{
+		[JsonProperty("old_index")]
+		string OldIndex { get; }
```
Comment (russcam) [inline src/Nest/Indices/IndexManagement/RolloverIndex/RolloverIndexResponse.cs]: `IndexName`?

File: src/Nest/Indices/IndexManagement/RolloverIndex/RolloverConditions.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+	[JsonObject(MemberSerialization.OptIn)]
+	[JsonConverter(typeof(ReadAsTypeJsonConverter<RolloverConditions>))]
+	public interface IRolloverConditions
+	{
+		[JsonProperty("max_age")]
+		Time MaxAge { get; set; }
+
+		[JsonProperty("max_docs")]
+		int MaxDocs { get; set; }
```
Comment (russcam) [inline src/Nest/Indices/IndexManagement/RolloverIndex/RolloverConditions.cs]: [this is defined as a `long` in Elasticsearch](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/action/admin/indices/rollover/MaxDocsCondition.java#L31)

**Comment by russcam** [comment]
One small comment, but otherwise LGTM :+1:

File: src/Nest/Indices/IndexManagement/RolloverIndex/RolloverConditions.cs (hunk 2 above)
Comment (gmarz) [inline src/Nest/Indices/IndexManagement/RolloverIndex/RolloverConditions.cs]: Nice catch :+1:

**Pull request merged**